### PR TITLE
Pub 1005 Upgrade log4j2 version to 2.17.1

### PR DESCRIPTION
**Pull request by junaidiqbalmoj**
**Before creating a pull request make sure that:**

- [x] commit messages are meaningful and follow good commit message guidelines

Upgrade log4j2 version to 2.17.1 to fix vulnerability issue.

Please remove this line and everything above and fill the following sections:

https://tools.hmcts.net/jira/browse/PUB-1005

**Does this PR introduce a breaking change?** (check one with "x")
[ ] Yes
[x] No

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -3,7 +3,7 @@ plugins {
   id 'checkstyle'
   id 'io.spring.dependency-management' version '1.0.11.RELEASE'
   id 'jacoco'
-  id 'org.springframework.boot' version '2.5.5'
+  id 'org.springframework.boot' version '2.5.7'
```
Comment (kurt-uk) [inline build.gradle:6]: https://tools.hmcts.net/jira/secure/RapidBoard.jspa?rapidView=1608&projectKey=PUB&view=detail&selectedIssue=PUB-1005#:~:text=To%20upgrade%20code%20repositories%20to%20use%202.6.2

please see ticket for upgrading

**Review by kurt-uk — CHANGES_REQUESTED** [review]

File: build.gradle (hunk 1 above)
Comment (junaidiqbalmoj) [reply build.gradle]: It was updated in master branch but not in pre-release. Confirmed it with @ChrisS1512.

**Review by furnivall — APPROVED** [review]
Nice one man. 👍

**Review by alberto-hortiguela — APPROVED** [review]
LGTM

**Review by kurt-uk — APPROVED** [review]

**Pull request closed without merging**